app/models/machine.rb

```diff
@@ -1,5 +1,11 @@
 class Machine < ApplicationRecord
-  validates_presence_of :location
-  
   belongs_to :owner
+  has_many :machine_snacks
+  has_many :snacks, through: :machine_snacks
+
+  validates_presence_of :location
+
+  def avg_snack_price
+    snacks.average(:price).to_f.round(2)
```

Comment:
You could also save the prices in your db as floats so you don't need to use `to_f` here. Also, keep any presentation logic in the views. `.round` is presentation logic in this case because it just has to do with how we want to present the average in the view.